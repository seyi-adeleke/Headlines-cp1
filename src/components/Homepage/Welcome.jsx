import React from 'react';
import { Link } from 'react-router';

import SelectNewsSource from '../Body/SelectNewsSource.jsx';
import actionArticles from '../../Actions/actions';


/**
 * @export  Welcome
 * @class Welcome
 * @extends {React.Component}
 */
class Welcome extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      user: JSON.parse(localStorage.getItem('user')).info.name,
    };
    this.getNews = this.getNews.bind(this);
  }
  /**
   * passes the source value to the action
   * @returns  true
   * @memberof Welcome
   */
  getNews() {
    localStorage.setItem('key', this.state.source);
    if (this.ref) {
      actionArticles.receiveDetails(this.state.source, 'top');
    }
    return true;
  }

  newSource(newState) {
    this.setState({
      source: newState,
    });
    return true;
  }

  render() {
    return (
      <div>
        <p className="text-center">Welcome, {this.state.user} </p>
        <SelectNewsSource
          ref={(c) => { this.ref = c; }}
          getSource={(newState, sortAvailable) => this.newSource(newState, sortAvailable)}
        />
        {this.state.source ?
        <div className="col-md-4" style={{ marginTop: 20 }}>
          <Link to={'headlines'}><button
            onClick={this.getNews}
            className="btn search-btn"
          >
            <b>Get News</b>
          </button>
          </Link>
        </div> : null}
        <br />
        <br />
      </div>
    );
  }
}
export default Welcome;
