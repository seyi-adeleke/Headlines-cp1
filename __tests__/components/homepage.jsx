import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';

import Homepage from '../../src/components/HomePage.jsx';

jest.mock('../../src/components/HomePage.jsx');


describe('Homepage Component', () => {
  it('should match the Homepage snapshot', () => {
    const component = mount(<Homepage />);
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
