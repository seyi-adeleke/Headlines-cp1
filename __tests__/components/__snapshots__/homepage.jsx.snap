// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage Component should match the Homepage snapshot 1`] = `<Homepage />`;
